utils/src/ooinstall/oo_config.py

```diff
@@ -236,9 +241,36 @@ def _upgrade_legacy_config(self):
         self.settings['variant'] = 'openshift-enterprise'
         self.settings['variant_version'] = '3.0'
 
-    def _upgrade_v1_config(self):
-        #TODO write code to upgrade old config
-        return
+    def _upgrade_v1_config(self, config):
+        new_config_data = {}
+        new_config_data['deployment'] = {}
+        new_config_data['deployment']['hosts'] = []
+        new_config_data['deployment']['roles'] = {}
+        new_config_data['deployment']['variables'] = {}
+
+        role_list = {}
+
+        if config.get('ansible_ssh_user', False):
+            new_config_data['deployment']['ansible_ssh_user'] = config['ansible_ssh_user']
+
+        for host in config['hosts']:
+            host_props = {}
+            host_props['roles'] = []
+            host_props['connect_to'] = host['connect_to']
+
+            for prop in ['ip', 'public_ip', 'hostname', 'public_hostname', 'containerized', 'preconfigured']:
+                host_props[prop] = host.get(prop, None)
+
+            for role in ['master', 'node', 'master_lb', 'storage', 'etcd']:
+                if host.get(role, False):
```

Comment:
@smunilla if we were just making assumptions, then I wouldn't expect the migration to work, since the etcd group wouldn't be populated...  or do we continue to make those assumptions with the v2 format if they are not present?